File: wantedmarket/src/main/java/org/example/wantedmarket/dto/product/ProductDetailDto.java
package org.example.wantedmarket.dto.product;

import lombok.*;
import org.example.wantedmarket.dto.order.OrderInfoDto;
import org.example.wantedmarket.status.ProductStatus;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ProductDetailDto {

    private Long productId;
    private String name;
    private Integer price;
    private ProductStatus status;
    private Long sellerId;
    private List<OrderInfoDto> transactionList = new ArrayList<>();

}

File: src/main/kotlin/com/api/jellomarket/dto/product/ProductDetailDTO.kt
package com.api.jellomarket.dto.product

data class ProductDetailDTO(
    val id: Long,
    val name: String,
    val price: Int,
    val state: String,
    val stock: Int,
    val description: String,
    val imageUrl: String,
    val sellerId: Long,
    val createdAt: String,
    val updatedAt: String,
)
